13-TeV-examples/uproot_python/ttbar_analysis.ipynb

```diff
@@ -0,0 +1,1236 @@
+        "* Combining Mass and Weight Lists\n",
+        "* Plotting\n",
+        "\n",
+        "\n",
+        "# **Running a Jupyter notebook**\n",
+        "To run the whole Jupyter notebook, in the top menu click Cell -> Run All.\n",
+        "\n",
+        "To propagate a change you've made to a piece of code, click Cell -> Run All Below.\n",
+        "\n",
+        "You can also run a single code cell, by clicking Cell -> Run Cells, or using the keyboard shortcut Shift+Enter.\n"
+      ],
+      "metadata": {
+        "id": "v9CDWygzhxRE"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "!pip install uproot"
+      ],
+      "metadata": {
+        "id": "63X2lXpgywvZ",
+        "collapsed": true
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# **To setup**\n",
+        "Cell -> Run All Below\n",
+        "\n",
+        "to be done every time you re-open this notebook\n",
+        "\n",
+        "We're going to be using a number of tools to help us:\n",
+        "\n",
+        "* uproot: lets us read .root files typically used in particle physics into data formats used in python\n",
+        "* awkward: lets us use efficiently the nested data in columnar format\n",
+        "* pandas: lets us store data as dataframes, a format widely used in python\n",
+        "* numpy: provides numerical calculations such as histogramming\n",
+        "* matplotlib: common tool for making plots, figures, images, visualisations\n",
+        "lmfit: tool for statistical fitting"
+      ],
+      "metadata": {
+        "id": "f477h5dJ-SFM"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import uproot  # For reading ROOT files efficiently\n",
+        "\n",
+        "import awkward as ak  # To represent nested data in columnar format\n",
+        "import numpy as np  # For numerical calculations such as histogramming\n",
+        "import pandas as pd\n",
+        "\n",
+        "\n",
+        "import matplotlib.pyplot as plt  # For creating plots and visualizations\n",
+        "from matplotlib.ticker import AutoMinorLocator  # For adding minor ticks to plot axes\n",
+        "\n",
+        "import requests  # For making HTTP requests\n",
+        "\n",
+        "import time  # For timing operations and adding delays if needed"
+      ],
+      "metadata": {
+        "id": "y7inFXIrVWMI"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# **Explanation of Key Parameters in the $t\\bar{t}$ Analysis Code**\n",
+        "\n",
+        "In particle physics analyses, various parameters and constants are essential for accurately processing data and interpreting results. Below is an explanation of the key parameters used in the\n",
+        " $t\\bar{t}$ analysis code provided:\n",
```

Comment:
We can add a link should the user be interested in more: https://atlas.cern/glossary